Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py

```diff
@@ -551,6 +558,621 @@ def decode_url(headers: str) -> dict:
+
+        except ValueError:
+            if res.text:
+                error = '\n{}'.format(res.text)
+            else:
+                error = "Received empty response from the server"
+
+        api_call_info = (
+            'Parameters used:\n'
+            '\tURL: {xsiam_url}\n'
+            '\tHeaders: {headers}\n\n'
+            'Response status code: {status_code}\n'
+            'Error received:\n\t{error}'
+        ).format(xsiam_url=xsiam_url, headers=json.dumps(headers, indent=8), status_code=res.status_code, error=error)
+
+        demisto.error(header_msg + api_call_info)
+        raise DemistoException(header_msg + error, DemistoException)
+
+    client = BaseClient(base_url=xsiam_url, proxy=add_proxy_to_request)
+    if data_size_expected_to_split_evenly:
+        data_chunks = split_data_to_chunks_evenly(data, chunk_size)
+    else:
+        data_chunks = split_data_to_chunks(data, chunk_size)
+
+    def send_events(data_chunk):
+        chunk_size = len(data_chunk)
+        data_chunk = '\n'.join(data_chunk)
+        zipped_data = gzip.compress(data_chunk.encode('utf-8'))  # type: ignore[AttributeError,attr-defined]
+        xsiam_api_call_with_retries(client=client, events_error_handler=data_error_handler,
+                                    error_msg=header_msg, headers=headers,
+                                    num_of_attempts=num_of_attempts, xsiam_url=xsiam_url,
+                                    zipped_data=zipped_data, is_json_response=True, data_type=data_type)
+        return chunk_size
+
+    async def send_events_async(data_chunk):
+        chunk_size = len(data_chunk)
+        data_chunk = '\n'.join(data_chunk)
+        zipped_data = gzip.compress(data_chunk.encode('utf-8'))  # type: ignore[AttributeError,attr-defined]
+        _ = await xsiam_api_call_async_with_retries(headers=headers,
+                                                    num_of_attempts=num_of_attempts, xsiam_url=xsiam_url,
+                                                    zipped_data=zipped_data, data_type=data_type)
+        return chunk_size
+
+    if send_events_asynchronously:
+        demisto.info(f"Running in interval = {counter}. Sending events to xsiam asynchronously.")
+        all_chunks = list(data_chunks)
+        demisto.info(f"Running in interval = {counter}. Finished appending all data_chunks to a list.")
+        tasks = [asyncio.create_task(send_events_async(chunk)) for chunk in all_chunks]
+
+        demisto.info(f'Finished submiting {len(tasks)} tasks for the {counter} time')
+        return tasks
+    else:
+        demisto.info("Sending events to xsiam synchronously.")
+        for chunk in data_chunks:
+            data_size += send_events(chunk)
+
+        if should_update_health_module:
+            demisto.updateModuleHealth({'{data_type}Pulled'.format(data_type=data_type): data_size})
+    return
+
+
+def split_data_to_chunks_evenly(data, target_chunk_size):  # pragma: no cover
```

Comment:
not a great name (since both of them will be even and this one actually theres more of a risk that it isnt.
Lets specify split by slice or something like that